test/src/userobjects/TestShapeElementUserObject.C

```diff
@@ -0,0 +1,90 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "TestShapeElementUserObject.h"
+
+template<>
+InputParameters validParams<TestShapeElementUserObject>()
+{
+  InputParameters params = validParams<ShapeElementUserObject>();
+  params.addCoupledVar("u", "first coupled variable");
+  params.addRequiredParam<unsigned int>("u_dofs", "Number of degrees of freedom per element for u");
+  params.addCoupledVar("v", "second coupled variable");
+  params.addRequiredParam<unsigned int>("v_dofs", "Number of degrees of freedom per element for v");
+  return params;
+}
+
+TestShapeElementUserObject::TestShapeElementUserObject(const InputParameters & parameters) :
+    ShapeElementUserObject(parameters),
+    _u_var(coupled("u")),
+    _u_dofs(getParam<unsigned int>("u_dofs")),
+    _v_var(coupled("v")),
+    _v_dofs(getParam<unsigned int>("v_dofs"))
+{
+}
+
+void
+TestShapeElementUserObject::initialize()
+{
+  _execute_mask = 0;
+}
+
+void
+TestShapeElementUserObject::execute()
+{
+}
+
+void
+TestShapeElementUserObject::executeJacobian(unsigned int jvar)
+{
+  // derivative of _integral w.r.t. u_j
+  if (jvar == _u_var)
+  {
+    // internal testing to make sure the _phis are initialized, set flag on call
+    if (_phi.size() != _u_dofs)
+      mooseError("Shape functions for u are initialized incorrectly. Expected " << _u_dofs << " DOFs, found " << _phi.size());
+    _execute_mask |= 1;
```

Comment:
My current PR allows you to use bit operations on C++ enums which makes this kind of code way more readable.

e.g.
```
Foo::_execute_mask |= Foo::JACOBIAN;
```